UIforETW/Utility.cpp

```diff
@@ -298,6 +298,47 @@ std::wstring ConvertToCRLF(const std::wstring& input)
 	return result;
 }
 
+std::wstring ReadRegistryString(HKEY root, const std::wstring& subkey, const std::wstring& valueName, bool force32Bit)
+{
+	std::wstring value;
+	const DWORD flags = RRF_RT_REG_SZ | RRF_RT_REG_EXPAND_SZ | RRF_ZEROONFAILURE;
+
+	REGSAM openOptions = KEY_QUERY_VALUE;
+	if (force32Bit)
+	{
+		openOptions |= KEY_WOW64_32KEY;
+	}
+
+	HKEY key;
+	if (::RegOpenKeyExW(root, subkey.c_str(), 0, openOptions, &key) != ERROR_SUCCESS)
+	{
+		return value;
+	}
+
+
+	DWORD bufSize = 50 * sizeof(wchar_t);
+	LSTATUS result = ERROR_MORE_DATA;
+	while (result == ERROR_MORE_DATA)
+	{
+		value.resize(bufSize / sizeof(wchar_t));
+		DWORD type;
+		result = ::RegGetValueW(key, nullptr, valueName.c_str(), flags, &type, const_cast<wchar_t*>(value.data()), &bufSize);

```

Comment:
This introduces a new CoreGuidelines analysis warning about using `const_cast`. It's not needed in C++17 (or latest) mode, but I didn't feel comfortable changing the build requirements for the project. But since *eventually* the `const_cast` isn't needed, it felt okay to do.